### PR TITLE
New feature: posibility to send all users a custom mail.

### PR DESCRIPTION
**Pull request by danielgimeno**
Dev: From users index now you can select some and then send them a custom mail.
Dev: It's possible to send a single user an email.

### CODE REVIEW

**Comment by SamMousa** [comment]
I don't think this should be merged. 
1. It is a totally new feature, 2.5 is feature locked.
2. It introduces new functions in the global namespace which is bad practice. 
3. It implements a new static function on the user class that returns a list in some specific format; formatting should not be done in the model layer.
4. It contains tmp/runtime.


On 4 i'm 100% sure @c-schmitz will agree, on the others you might get lucky ;-)

**Comment by Shnoulle** [comment]
+1 : must be in plugin. If you don't find the way to have a plugin like this : put a feature request for a new event (actuallyi don't see any part needed, except for "more easyli" : add a menu / easily add a page.

And more : 
- inline style
- inline script

**Comment by c-schmitz** [comment]
I agree to all mentioned points by sammousa

File: application/controllers/admin/useraction.php [hunk 1]
```diff
@@ -186,6 +186,91 @@ public function adduser()
     }
 
     /**
+    * Send email to LimeSurvey selected users
+    *
+    */
+    function sendMailToUser()
```
Comment (olleharstedt) [inline application/controllers/admin/useraction.php]: 1. Should have public/private/protected.
2. No function should be longer than 60 lines, for readability. Please factor out in smaller functions.

File: application/models/User.php [hunk 2]
```diff
@@ -292,4 +292,51 @@ public function relations()
...
 		);
 	}
+
+    /**
+    * Returns Users from string id list
+    *
+    * @access public
+    * @return array User list
+    */
+    public static function getArrayUsers($stringUsersList, $outputformat='fullinfoarray')
```
Comment (olleharstedt) [inline application/models/User.php]: Please add documentation for all function arguments.
Example:
`@param string $stringUsersList Comma-separated list with user ids`

**Comment by maziminke** [comment]
I really like the idea of having a feature to send an email to a certain set up users.

Is it possible to upload 2-3 screen shots of how the interface looks like?

If you are able to make this a plugin that also works with Limesurvey 2.06 I may be able to partly sponsor adjusting this feature to make it a plugin.

I assume c_schmitz doesn't mind adding the required plugin events to LS 2.50 and 2.06LTS if the rest of the code is clearly separated at a plugin?

**Comment by Shnoulle** [comment]
I really think this feature can be done without new event in plugin. 

1. Add settins : informarion (info) / user(select) / subject(string) / body (html|text) / send action (button)
2. With getPluginSettings public function : update info (according to next system) + user : update optioin with the list of user
3. With saveSettings public function : send email according to : button + user selected + subject + body . Add a plugin private var to fill the "information" part with getPluginSettings

Denis

**Comment by LouisGac** [comment]
I agree with mazi, those kind of functionalities are really very common in any webapp,  CMS, etc.
IMHO: Should be a core feature .

**Pull request closed without merging**